test/integration/resultsPage.js

```diff
@@ -0,0 +1,119 @@
+const utils = require('../lib/testUtils');
+
+const expect = chai.expect;
+
+chai.use(chaiHttp);
+
+const resultsRoute = `${constants.siteRoot}/results/`;
+
+function setTypeAndOriginPairs() {
+  return [
+    {
+      origin: constants.serviceChoices.symptoms,
+      type: constants.serviceTypes.professional,
+    },
+    {
+      origin: constants.serviceChoices.under16,
+      type: constants.serviceTypes.professional,
+    },
+    {
+      origin: constants.serviceChoices['16to25'],
+      type: constants.serviceTypes.professional,
+    },
+    {
+      origin: constants.serviceChoices.over25,
+      type: constants.serviceTypes.professional,
+    },
+    {
+      origin: constants.serviceChoices['16to25'],
+      type: constants.serviceTypes.kit,
+    },
+    {
+      origin: constants.serviceChoices.over25,
+      type: constants.serviceTypes.kit,
+    },
+  ];
+}
+
+function assertSearchResponse(location, done, assertions) {
+  setTypeAndOriginPairs().forEach((queryParams) => {
+    const origin = queryParams.origin;
+    const type = queryParams.type;
+    chai.request(app)
+      .get(resultsRoute)
+      .query({ location, origin, type })
+      .end((err, res) => {
+        expect(err).to.equal(null);
+        iExpect.htmlWith200Status(res);
+        assertions(err, res);
+      });
+  });
+  done();
+}
+
+describe('Results page results', function test() {
+  // Setting this timeout as it is calling the real DB...
+  this.timeout(utils.maxWaitTimeMs);
+
+  before((done) => {
+    utils.waitForSiteReady(done);
+  });
+
+  const location = 'ls1';
+
+  describe('layout', () => {
+    it('should contain HTML', (done) => {
+      assertSearchResponse(location, done, (err, res) => {
+        iExpect.htmlWith200Status(res);
+      });
+    });
+  });
+
+  describe('Each service', () => {
+    it('should have distance, a name, address and telephone number, a map link, See opening times toggle and See service information toggle', (done) => {
+      assertSearchResponse(location, done, (err, res) => {
+        const $ = cheerio.load(res.text);
+        const searchResultsDistance = $('.results__address.results__address-distance');
+        const searchResultsName = $('.results__name');
+        const searchResultsAddress = $('.results__address.results__address-lines');
+        const searchResultsPhone = $('.results__address.results__telephone a');
+        const searchResultsMapLink = $('.results__item__link a');
+        const searchResultsOpeningTimes = $('.results__item__opening-times summary span');
+        const searchResultsService = $('.results__item__service-details summary span');
+
+        expect(searchResultsDistance).to.not.equal(undefined);
+        expect(searchResultsName).to.not.equal(undefined);
+        expect(searchResultsAddress).to.not.equal(undefined);
+        expect(searchResultsPhone).to.not.equal(undefined);
+        expect(searchResultsMapLink.text()).to.contain('See map and directions');
```

Comment:
It looks like there isn't any additional text, does the `contain` need to be here or could it be `equal`?

Reply:
There's screen reader text, so contains is correct.